demo-app/src/app/ot/config-builder/page.tsx

```diff
@@ -0,0 +1,242 @@
+          <div key={option}>
+            <label htmlFor={option}>{option}</label>
+            <input
+              type="radio"
+              id={option}
+              name={option}
+              value={option}
+              checked={value === option}
+              onChange={(e) => {
+                onChange(e.target.value)
+              }}
+            />
+          </div>
+        ))
+        return <>{enumOptions}</>
+      }
+  }
+
+  return (
+    <div>
+      Name: {name}. type: {JSON.stringify(type)}
+    </div>
+  )
+}
+
+function PropInputs({
+  componentName,
+  onChange,
+  props,
+}: {
+  componentName: string
+  onChange: (propName: string) => (value: YamlType) => void
+  props: ViewProps
+}): React.JSX.Element | null {
+  const component = ALL_COMPONENTS[componentName]
+  if (!hasPropsExport(component)) {
+    return null
+  }
+  const niceProps = describeZod(component.Props.shape)
+
+  return (
+    <div>
+      {Object.entries(niceProps).map(([propName, propType]) => {
+        return (
+          <PropInput
+            key={propName}
+            onChange={onChange(propName)}
+            name={propName}
+            type={propType}
+            value={props ? (props[propName] as string) : undefined}
+          />
+        )
+      })}
+    </div>
+  )
+}
+
+function hasPropsExport(
+  component: any,
+): component is OpenTrussComponentExports {
+  return 'Props' in component
+}
+
+function ComponentListItem({
+  componentName,
+}: {
+  componentName: string
+}): React.JSX.Element {
+  const { config, setConfig } = useContext(ConfigBuilderContext)
+  const [props, setProps] = useState<ViewProps>(undefined)
+  const parsedConfig = parseYaml(config) as unknown as WorkflowSpec
+
+  const frame: FrameType = {
+    frame: null, // Makes the config easier to read
+    view: { component: componentName, props },
+  }
+  const addFrame = (): void => {
+    const frames = (parsedConfig.workflow.frames || []).concat(frame)
+    setConfig(
+      stringifyYaml({
+        ...parsedConfig,
+        workflow: { ...parsedConfig.workflow, frames },
+      }),
+    )
+  }
+  const onChange = (propName: string) => (value: YamlType) => {
+    setProps((currentProps) => ({ ...currentProps, [propName]: value }))
+  }
+
+  return (
+    <div>
+      {componentName}
+      <button onClick={addFrame}>Add</button>
+      <PropInputs
+        componentName={componentName}
+        onChange={onChange}
+        props={props}
+      />
+      <br />
+      <br />
+    </div>
+  )
+}
+
+function ComponentList(): React.JSX.Element {
+  return (
+    <div>
+      {Object.keys(ALL_COMPONENTS).map((componentName) => (
+        <ComponentListItem key={componentName} componentName={componentName} />
+      ))}
+    </div>
+  )
+}
+
+export default function ConfigBuilderPage(): React.JSX.Element {
+  const [config, setConfig] = useState<string>(CONFIG_BASE)
+
+  return (
+    <ConfigBuilderContext.Provider value={{ config, setConfig }}>
```

Comment:
Can you see a future where this entire page is built as an OT component? I wonder if / when we solve the state problem (e.g. with jotai / signals) we can obviate the need to create this context. What do you think?

Anyhow, 👍 with building it the way you did for now so we don't have to boil the ocean to get this shipped.